pkg/apis/apps/v1/conversion.go

```diff
@@ -453,8 +453,10 @@ func Convert_apps_StatefulSetUpdateStrategy_To_v1_StatefulSetUpdateStrategy(in *
 }
 
 func Convert_v1_StatefulSetStatus_To_apps_StatefulSetStatus(in *appsv1.StatefulSetStatus, out *apps.StatefulSetStatus, s conversion.Scope) error {
-	out.ObservedGeneration = new(int64)
-	*out.ObservedGeneration = in.ObservedGeneration
+	if in.ObservedGeneration != nil {
```

Comment:
This does not make sense.
`in. ObservedGeneration` is `int64`, which cannot be `nil`.
I prefer to keep this unchanged.

Reply:
@dixudx [v1beta1](https://github.com/kubernetes/kubernetes/blob/091794c0874bed681cdf221dd7fbc8f31dbeb0b1/staging/src/k8s.io/api/apps/v1beta1/types.go#L222) still uses *int64.  It is inconsistent across api versions which one should we follow?

Reply:
Refer to #49607. Don't change the type back.